lib/Frontend/ArgsToFrontendOptionsConverter.cpp

```diff
@@ -149,6 +147,15 @@ bool ArgsToFrontendOptionsConverter::convert(
     Opts.RequestedAction = determineRequestedAction(Args);
   }
 
+  if (Opts.RequestedAction == FrontendOptions::ActionType::CompileModuleFromInterface) {
+    // The situations where we use this action, e.g. explicit module building and
+    // generating prebuilt module cache, don't need synchronization. We should avoid
+    // using lock files for them.
```

Comment:
What is the action for lazy module interface builds? It would be the only case where we would need the lock, is this right?

Reply:
That's correct. We only need lock file for regular compilation modes like `-emit-module` and `-merge-modules` because several compiler processes may attempt to build the same module.

Reply:
Aren't the sub invocations of the compiler for lazy module interface builds using the same `-compile-module-from-interface` flag? I'm worried this is disabling the lock even in other modes.

Reply:
Lazy module interface builds won't spawn a new front-end job to compile module from an interface file. Instead, the module building step happens implicitly in the main compilation action like `-emit-module`. We are moving towards explicitly module building where `-compile-module-from-interface` flag will be used, but that's where the lock file isn't necessary because all actions are performed synchronously.

Reply:
Alright, thanks for the explanation!